schematic/models/metadata.py

```diff
@@ -235,3 +237,45 @@ def populateModelManifest(self, title, manifestPath: str, rootNode: str) -> str:
         emptyManifestURL = mg.get_manifest()
 
         return mg.populate_manifest_spreadsheet(manifestPath, emptyManifestURL)
+
+
+    def submit_metadata_manifest(self, manifest_path: str, dataset_id: str, validate_component: str, json_schema: str = None) -> bool:
+        """Wrap methods that are responsible for validation of manifests for a given component, and association of the 
+        same manifest file with a specified dataset.
+
+        Args:
+            manifest_path: Path to the manifest file, which contains the metadata.
+            dataset_id: Synapse ID of the dataset on Synapse containing the metadata manifest file.
+            validate_component: Component from the schema.org schema based on which the manifest template has been generated.
+
+        Returns:
+            True: If both validation and association were successful.
+            False: If the function could not successfully validate (and hence associate) the manifest file with the dataset.
+        """
+        # list of all the validation errors present in the manifest file
+        val_errors = []
+
+        # JSON schema need not be provided always, it is an optional argument
+        # if JSON schema is provided, then use it
+        if json_schema:
+            val_errors = self.validateModelManifest(manifestPath=manifest_path, rootNode=validate_component, jsonSchema=json_schema)
+
+        # automatic JSON schema generation and validation with that JSON schema
+        val_errors = self.validateModelManifest(manifestPath=manifest_path, rootNode=validate_component)
+
+        # create object of SynapseStorage() class to associate manifest file with dataset
+        syn = synapseclient.Synapse(configPath=CONFIG.SYNAPSE_CONFIG_PATH)
```

Comment:
Should we allow Synapse-specific code to (further?) creep outside of `synapse/store.py`? It might be worth considering how to tweak this function's  signature (and/or internals) to better accommodate different asset stores.

Reply:
I agree with @BrunoGrandePhD here - can we implement everything related to synapse in a synapse/store method and just call it here, using the store module as an interface?

Reply:
Right, so we're creating an instance of a `synapseclient.Synapse()` object, using it to login and passing it to an instance of `SynapseStorage`. Since we're not "tweaking"/modifying any `SynapseStorage` class functionalities, I'm not sure it fits in `synapse/store.py`? Perhaps a `utils/synapse_utils.py`?

Reply:
Yes, let's move these two lines of code in the else-part of the SynapseStorage class constructor (we actually had that pathway/option in synapse/store module when working on schematic initially).